libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -617,21 +650,20 @@ bool AC_PrecLand::construct_pos_meas_using_rangefinder(float rangefinder_alt_m,
             }
             if (_backend->distance_to_target() > 0.0f) {
                 // sensor has provided distance to landing target
-                dist = _backend->distance_to_target();
-                alt = dist * target_vec_unit_ned.z;
+                dist_to_target = _backend->distance_to_target();
             } else {
                 // sensor only knows the horizontal location of the landing target
                 // rely on rangefinder for the vertical target
-                alt = MAX(rangefinder_alt_m - cam_pos_ned.z, 0.0f);
```

Comment:
I think realistically, no one will ever use a forward-facing RangeFinder for docking (unless we assume that the docking point has a large wall behind it). I think it is very safe to assume that we expect the users to send in the "distance to the target" with the mavlink message (LANDING_TARGET) for docking to be usable.
A suggestion: Why don't we leave the rangefinder stuff to be Copter only (since that's the only place it's actually going to be used)? For Rovers, this library only works if DISTANCE is sent with the sensor. This will make the code a lot simpler, we'll avoid unnecessary calculations.

Reply:
The only issue then is that IR-Lock becomes unusable for this Rover. I don't think anyone would be able to use IR-Lock anyway for docking... You could compile it out for Rover

Reply:
These are just my thoughts.. I don't feel too strongly about it. I would just prefer that the code remains as simple as possible

Reply:
I'm not sure we should make changes to disable the use of a rangefinder + IRLock because introducing differences between Copter and Rover adds the risk of bugs.